src/functionaltests/java/com/ericsson/ei/threadingAndWaitlistRepeat/ThreadingAndWaitlistRepeatSteps.java

```diff
@@ -80,45 +75,46 @@ public void no_event_is_aggregated() throws Throwable {
         assertEquals("aggregatedObjectExists was true, should be false, ", false, aggregatedObjectExists);
     }
 
-    @Then("^the waitlist will try to resend the events at given time interval$")
-    public void the_waitlist_will_try_to_resend_the_events_at_given_time_interval() throws Throwable {
-        TimeUnit.SECONDS.sleep(5);
-        List<String> resentEvents = new ArrayList<>();
-        List<String> lines = new ArrayList<>(Files.readAllLines(tempLogFile.toPath()));
-
-        int waitlistId = waitListWorker.hashCode();
-        String waitListIdPattern = "FROM WAITLIST: " + waitlistId;
-        Pattern pattern = Pattern.compile("\\[EIFFEL EVENT RESENT " + waitListIdPattern + "\\] id:([a-zA-Z\\d-]+)");
-        for (String line : lines) {
-            Matcher matcher = pattern.matcher(line);
-            if (matcher.find() && !matcher.group(1).equals("")) {
-                if (!resentEvents.contains(matcher.group(1))) {
-                    resentEvents.add(matcher.group(1));
-                }
-            }
+    @Then("^event-to-object-map is manipulated to include the sent events$")
+    public void event_to_object_map_is_manipulated_to_include_the_sent_events() throws Throwable {
+        JsonNode parsedJSON = eventManager.getJSONFromFile(EIFFEL_EVENTS_JSON_PATH);
+        ObjectMapper objectMapper = new ObjectMapper();
+        rulesJson = objectMapper.readTree(ID_RULE);
+        rulesObject = new RulesObject(rulesJson);
+
+        String dummyObjectID = "1234abcd-12ab-12ab-12ab-123456abcdef";
+        List<String> eventNames = getEventNamesToSend();
+        for (String eventName : eventNames) {
+            JsonNode eventJson = parsedJSON.get(eventName);
+            eventToObjectMapHanler.updateEventToObjectMapInMemoryDB(rulesObject, eventJson.toString(), dummyObjectID);
         }
-        assertEquals(getEventNamesToSend().size(), resentEvents.size());
+    }
+
+    @Then("^when waitlist has resent events they should have been deleted$")
+    public void when_waitlist_has_resent_events_they_should_have_been_deleted() throws Throwable {
+        TimeUnit.SECONDS.sleep(2);
```

Comment:
Why a sleep?
Isn't it better to poll DB until all events has been deleted in the test. And if it takes too long time, use a timeout that interrupt the test.